test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/TestBadFormat.java

```diff
@@ -623,6 +624,74 @@ public void runInvalidReuse1() {}
     public void runInvalidReuse2() {}
 }
 
+class BadSetupTest {
+    // ----------- Bad Combinations of Annotations -----------------
+    @Setup
+    @Test
+    public Object[] badSetupTestAnnotation() {
+      return new Object[]{1, 2, 3};
+    }
+
+    @NoFail
+    @Test
+    public void testForBadSetupCheckAnnotation() {}
+
+    @Setup
+    @Check(test = "testForBadSetupCheckAnnotation")
+    public void badSetupCheckAnnotation() {}
+
+    @Setup
+    @Arguments(values = {Argument.NUMBER_42, Argument.NUMBER_42})
+    public void badSetupArgumentsAnnotation(int a, int b) {}
+
+    @NoFail
+    @Test
+    public void testForBadSetupRunAnnotation() {}
+
+    @Setup
+    @Run(test = "testForBadSetupRunAnnotation")
+    public void badSetupRunAnnotation() {}
+
+    // ----------- Ok: Setup Without Test Method -----------------
```

Comment:
Maybe change this to "Useless but ok". Maybe we still want to add some verification for these useless methods. But let's do that separately.

Reply:
will do